hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetService.java

```diff
@@ -279,4 +291,15 @@ public void populate(LiveOperations liveOperations) {
         }
         return keys;
     }
+
+    private Thread shutdownHookThread(NodeEngine nodeEngine) {
+        return new Thread(() -> {
```

Comment:
would be good to give the thread a name

Reply:
"jet.ShutdownThread" ? imdg has "hz.ShutdownThread"

Reply:
Yes that's fine.